executor/src/witgen/jit/identity_queue.rs

```diff
@@ -132,20 +157,25 @@ where
 
 #[derive(Clone)]
 pub enum QueueItem<'a, T: FieldElement> {
+    Equation {
+        expr: QuadraticSymbolicExpression<T, Variable>,
+        /// If only variables whose numeric value is known at compile-time are considered
+        /// "known".
+        require_concretely_known: bool,
+        // TODO we could add here, which identity and which row this stems from.
+        // then we can re-use these in place of most of the witgen_inference::evaluate calls.
+    },
     Identity(&'a Identity<T>, i32),
-    VariableAssignment(VariableAssignment<'a, T>),
-    ConstantAssignment(ConstantAssignment<'a, T>),
     ProverFunction(ProverFunction<'a, T>, i32),
 }
 
 impl<T: FieldElement> Ord for QueueItem<'_, T> {
     fn cmp(&self, other: &Self) -> std::cmp::Ordering {
         match (self, other) {
+            // TODO comparison of Equation is not properly implemented here.
```

Comment:
I think we can remove the implementations for Ord, PartialOrd, Eq, PartialEq from `QueueItem`!

Reply:
ah no, you are right!

Reply:
This is still open though?

Reply:
Oh sorry, never mind, you fixed it